src/FSharp.Data.GraphQL.Server/Execution.fs

```diff
@@ -563,21 +563,28 @@ let private executeQueryOrMutation (resultSet: (string * ExecutionInfo) []) (ctx
                     let nvl (path : string list) data =
                         let path' = path |> List.map (fun p -> p :> obj)
                         NameValueLookup.ofList ["data", data; "path", upcast path']
+                    let window path size item =
+                        if size > 1
+                        then item |> Seq.windowed size |> Seq.map ((fun i -> i :> obj) >> (fun d -> nvl path d))
+                        else item
+                    let stream (path : string list) (batchSize : int) (value : obj) =
+                        match value with
+                        | :? NameValueLookup as x -> 
+                            x |> Seq.map (fun x -> 
+                                match x.Value with
+                                | :? IEnumerable<obj> as x -> x |> Seq.mapi (fun i d -> nvli path i d) |> window path batchSize
```

Comment:
When a batched stream happens, each single result is wrapped into another stream result, with no index.

For example, a stream "one-by-one" of two items would be something like:

```
{"data":[{"id":"1002","name":"Han Solo"}],"path":["hero","friends",0]}
{"data":[{"id":"1003","name":"Leia Organa"}],"path":["hero","friends",1]}
```

When streaming in batches of two items, we would receive this:

```
{"data":
  [{"data":[{"id":"1002","name":"Han Solo"}],"path":["hero","friends",0]},
   {"data":[{"id":"1003","name":"Leia Organa"}],"path":["hero","friends",1]}],
 "path":["hero","friends"]}
```